components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -1145,12 +1160,13 @@ get_var_dimnames (const FieldLayout& layout) const
 {
   strvec_t dims;
   for (int i=0; i<layout.rank(); ++i) {
-    const auto t = layout.tag(i);
+    const int ind = m_transpose ? layout.rank()-i-1 : i;
```

Comment:
`FieldLayout` has the method `transpose`, that returns the transposed layout. Perhaps you could transpose the layout at the call site, and avoid the index logic here (index logic is often confusing)
```c++
m_var_dims[fname] = get_var_dimnames(m_transpose ? layout.transpose() : layout);
```